### PR TITLE
code health sanitytest

### PR DESCRIPTION
**Pull request by pauli-oracle**
- [x] tests are added or updated

Update sanitytest to use the test infrastructure

### CODE REVIEW

File: test/sanitytest.c [hunk 1]
```diff
@@ -1,67 +1,92 @@
...
 
     /* Enum size */
-    TEST(sizeof(enum smallchoices) == sizeof(int));
-    TEST(sizeof(enum medchoices) == sizeof(int));
-    TEST(sizeof(enum largechoices) == sizeof(int));
+    if (!TEST_size_t_eq(sizeof(enum smallchoices), sizeof(int)))
+        return 0;
+    if (!TEST_size_t_eq(sizeof(enum medchoices), sizeof(int)))
+        return 0;
+    if (!TEST_size_t_eq(sizeof(enum largechoices), sizeof(int)))
```
Comment (mattcaswell) [inline test/sanitytest.c]: Single if

File: test/sanitytest.c [hunk 2]
```diff
@@ -1,67 +1,92 @@
...
+}
+
+static int test_sanity_twos_complement(void)
+{
     /* Basic two's complement checks. */
-    TEST(~(-1) == 0);
-    TEST(~(-1L) == 0L);
+    if (!TEST_int_eq(~(-1), 0))
+        return 0;
+    if (!TEST_long_eq(~(-1L), 0L))
```
Comment (mattcaswell) [inline test/sanitytest.c]: Single if...etc

File: test/sanitytest.c (hunk 1 above)
Comment (levitte) [inline test/sanitytest.c]: ... and preference toward breaking before operators (friendly reminder)

**Comment by pauli-oracle** [comment]
Changed to single ifs

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**